contrib/cirrus/runner.sh

```diff
@@ -269,6 +269,20 @@ msg "************************************************************"
 ((${SETUP_ENVIRONMENT:-0})) || \
     die "Expecting setup_environment.sh to have completed successfully"
 
+# Nearly every task in .cirrus.yml makes use of this shell script
+# wrapped by /usr/bin/time to collect runtime statistics.  Because the
+# --output option is used to log stats to a file, every child-process
+# inherits an open FD3 pointing at the log.  However, some testing
+# operations depend on making use of FD3, and so it must be explicitly
+# closed here (and for all further child-processes).
+if [[ -n "$STATS_LOGFILE" ]] && \
+    [[ "$(readlink /proc/self/fd/3)" == "$STATS_LOGFILE" ]]; then
+    warn "Closing FD 3 - open /usr/bin/time output log file"
```

Comment:
Could I suggest not bothering with the if/warn stuff, and just doing a plain `exec 3<&-`? I see no reason whatsoever to do any checks nor to inform the user.

Reply:
I think it's helpful in the case where the script is run manually after `hack/get_ci_vm.sh`.  It's also important because the cause (`/usr/bin/time --output`) is so physically distant from the mitigation.  I can easily envision a scenario where somebody tinkers with `.cirrus.yml` such that one of these warnings is the only clue as to missing expectations.  This then (hopefully) draws needed attention toward the comment.

Reply:
I humbly suggest that anyone playing with fd3 is either (a) very smart and can figure it out, or (b) very stupid and deserves what they get. Please don't add complexity for a super-rare case.

Reply:
> very smart and can figure it out

Normally I would agree, except my experience is that virtually none of the podman golang-devs even try to look at the bash scripts.  They look at the e2e test results, *maybe* the log output, then panic and "ask Chris".

> very stupid and deserves what they get

That doesn't strike me as very compassionate toward future maintainers.  They are by definition "stupid" when it comes to the automation scripts.  

---

Point of fact:  Since `exec 3<&-` always works, even when not open, adding conditional warnings are the only method available to provide _any_ clues as to what might be happening.  When it comes to providing *some* possibly helpful  information vs remaining silent, even for a rare (but detectable) case: It's easier to get noticed in log output, than require digging through unfamiliar automation scripts.

---

Where is the compromise solution-space?

What if I remove the conditionals, and just leave a persistent warning message that specifically references `/proc/self/fd/3` and `$STATS_LOGFILE`?

Other options?

Reply:
Please do what you think is right. My comments were based on my suspicion that there is no possible scenario in which anyone will ever want to run e2e tests with fd3 intentionally open, compounded by my dislike of unnecessary unmaintainable confusing complexity. I am stepping back now and will comment no further.

Reply:
I appreciate your feedback, and I respect we're both trying to "do the right thing".  I'm just not always certain what that is.

I've rebased and force-pushed w/ only a warning message here.  I'll peep at the log output, then give it a last think before I call this ready for final review...

Reply:
...re-pushed with shorter warning message, re-positioned so it appears right after Cirrus-CI prints the command-line (which includes `/usr/bin/time`).

Reply:
Hmmm...nope, seeing it in the logs now, I'm convinced you were right to suggest remaining silent.  Removing the warning message...